lib/spidr/agent.rb

```diff
@@ -338,6 +363,54 @@ def clear
       return self
     end
 
+
+    #
+    # Start spidering at a given URL.
+    #
+    # @param [String] html
+    #   The html to start spidering at.
+    #
+    # @yield [page]
+    #   If a block is given, it will be passed every page visited.
+    #
+    # @yieldparam [Page] page
+    #   A page which has been visited.
+    #
+    def start_with(html,&block)
+      response = Class.new do
```

Comment:
Also please no anonymous classes in methods. Better to create a Value Object style Class with a descriptive name.